streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```
@@ -655,7 +655,11 @@ void suspend(final boolean clean,
             }
         } else {
             // In the case of unclean close we still need to make sure all the stores are flushed before closing any
-            super.flushState();
+            try {
+                stateMgr.flush();
+            } catch (final ProcessorStateException e) {
+                // ignore any exceptions while flushing (all stores would have had a chance to flush anyway)
+            }
```

Judging from the stacktrace generated in the prior logging change, this is where we're throwing a TaskMigratedException, even though this block is an "unclean close". Just trying out ignoring the exception to see what happens.

---

Just to clarify we should only swallow under `dirty` suspension (in trunk it is wrapped in `close`) right? For clean close / suspend if it throws we would capture it and wrap as TaskMigrated still and then handle them by closing the task as dirty (in that second try we would swallow).

---

That's right